tests/transform/test_transform_points.py

```diff
@@ -70,3 +70,44 @@ def test_z_size_fail():
     emsg = "Cannot transform points, 'xs' and 'zs' require same length"
     with pytest.raises(ValueError, match=emsg):
         _ = transform_points(src_crs=WGS84, tgt_crs=WGS84, xs=data, ys=data, zs=zs)
+
+
+@pytest.mark.parametrize("zoffset", [None, 100])
+@pytest.mark.parametrize("reshape", [False, True])
+@pytest.mark.parametrize("roundtrip", [False, True])
+def test_transform(mocker, zoffset, reshape, roundtrip):
+    """Test transformation with identical and different CRSs."""
+    xs = np.arange(size := 10, dtype=float)
+    ys = np.arange(size, dtype=float) + 10
+    zs = np.arange(size, dtype=float) + zoffset if zoffset is not None else zoffset
+    if reshape:
+        shape = (2, 5)
+        xs, ys = xs.reshape(shape), ys.reshape(shape)
+        if zs is not None:
+            zs = zs.reshape(shape)
+    else:
+        shape = (size,)
+    shape = shape + (3,)
+    spy_from_crs = mocker.spy(Transformer, "from_crs")
+    spy_transform = mocker.spy(Transformer, "transform")
+    if roundtrip:
+        tmp = transform_points(src_crs=WGS84, tgt_crs="+proj=eqc", xs=xs, ys=ys, zs=zs)
+        result = transform_points(
+            src_crs="+proj=eqc",
+            tgt_crs=WGS84,
+            xs=tmp[..., 0],
+            ys=tmp[..., 1],
+            zs=tmp[..., 2],
+        )
```

Comment:
Small suggestion. `points` makes more sense to me.
```suggestion
        points = transform_points(src_crs=WGS84, tgt_crs="+proj=eqc", xs=xs, ys=ys, zs=zs)
        result = transform_points(
            src_crs="+proj=eqc",
            tgt_crs=WGS84,
            xs=points[..., 0],
            ys=points[..., 1],
            zs=points[..., 2],
        )
```